### PR TITLE
[Minimal-VM] Enable apparmor tests on TW and 15-SP4

### PR DESCRIPTION
**Pull request by jlausuch**
VRs:
TW: https://openqa.opensuse.org/tests/2257704
SLE15-SP4: https://openqa.suse.de/tests/8365124

### CODE REVIEW

**Comment by jlausuch** [comment]
@lilyeyes I see you as maintainer of [aa_autodep](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/security/apparmor/aa_autodep.pm), it is failing in JeOS in the step `ls -1 /usr/bin/pam*`. Do you know why this happens?

**Comment by lilyeyes** [comment]
> @lilyeyes I see you as maintainer of [aa_autodep](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/security/apparmor/aa_autodep.pm), [it is failing in JeOS](https://openqa.opensuse.org/tests/2257704#step/aa_autodep/29) in the step `ls -1 /usr/bin/pam*`. Do you know why this happens?

It seems there is no `/usr/bin/pam*` files found at that moment.
It might be a test case issue. Please file a poo to us to fix.

**Comment by jlausuch** [comment]
> > @lilyeyes I see you as maintainer of [aa_autodep](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/security/apparmor/aa_autodep.pm), [it is failing in JeOS](https://openqa.opensuse.org/tests/2257704#step/aa_autodep/29) in the step `ls -1 /usr/bin/pam*`. Do you know why this happens?
> 
> It seems there is no `/usr/bin/pam*` files found at that moment. It might be a test case issue. Please file a poo to us to fix.

I was looking at 15-SP4 job https://openqa.suse.de/tests/8346221#step/aa_autodep/23 where those files exist. I wonder why the difference is and if I am missing some configuration or package installation.

**Comment by lilyeyes** [comment]
> > > @lilyeyes I see you as maintainer of [aa_autodep](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/security/apparmor/aa_autodep.pm), [it is failing in JeOS](https://openqa.opensuse.org/tests/2257704#step/aa_autodep/29) in the step `ls -1 /usr/bin/pam*`. Do you know why this happens?
> > 
> > 
> > It seems there is no `/usr/bin/pam*` files found at that moment. It might be a test case issue. Please file a poo to us to fix.
> 
> I was looking at 15-SP4 job https://openqa.suse.de/tests/8346221#step/aa_autodep/23 where those files exist. I wonder why the difference is and if I am missing some configuration or package installation.

We met this issue on SLES before it might be performance or fs sync issue so we add "# Check the test files exist, double check to avoid performance issue" related code: see https://openqa.suse.de/tests/8346221/modules/aa_autodep/steps/1/src
Maybe this code doesn't work for JeOS.

**Comment by jlausuch** [comment]
> > > > @lilyeyes I see you as maintainer of [aa_autodep](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/security/apparmor/aa_autodep.pm), [it is failing in JeOS](https://openqa.opensuse.org/tests/2257704#step/aa_autodep/29) in the step `ls -1 /usr/bin/pam*`. Do you know why this happens?
> > > 
> > > 
> > > It seems there is no `/usr/bin/pam*` files found at that moment. It might be a test case issue. Please file a poo to us to fix.
> > 
> > 
> > I was looking at 15-SP4 job https://openqa.suse.de/tests/8346221#step/aa_autodep/23 where those files exist. I wonder why the difference is and if I am missing some configuration or package installation.
> 
> We met this issue on SLES before it might be performance or fs sync issue so we add "# Check the test files exist, double check to avoid performance issue" related code: see https://openqa.suse.de/tests/8346221/modules/aa_autodep/steps/1/src Maybe this code doesn't work for JeOS.

You mean we should wait a bit more until the `sync` command finishes syncing? 
I think it would be better to turn that double check into a loop with timeout. I'll try..

**Comment by jlausuch** [comment]
@lilyeyes I have checked the 15-SP4 test and this is the output I get BEFORE executing `aa-autodep -d /tmp/apparmor.d/ nscd`:

```
susetest:~ # ls -1 /usr/bin/pam*
/usr/bin/pamaddnoise
/usr/bin/pamarith
/usr/bin/pambackground
/usr/bin/pambayer
/usr/bin/pamchannel
/usr/bin/pamcomp
/usr/bin/pamcrater
/usr/bin/pamcut
/usr/bin/pamdeinterlace
/usr/bin/pamdepth
/usr/bin/pamdice
/usr/bin/pamditherbw
/usr/bin/pamedge
/usr/bin/pamendian
/usr/bin/pamenlarge
/usr/bin/pamexec
/usr/bin/pamfile
/usr/bin/pamfix
/usr/bin/pamfixtrunc
/usr/bin/pamflip
/usr/bin/pamfunc
/usr/bin/pamgauss
/usr/bin/pamgradient
/usr/bin/pamlookup
/usr/bin/pammasksharpen
/usr/bin/pammixinterlace
/usr/bin/pammosaicknit
/usr/bin/pamoil
/usr/bin/pampaintspill
/usr/bin/pamperspective
/usr/bin/pampick
/usr/bin/pampop9
/usr/bin/pamrecolor
/usr/bin/pamrgbatopng
/usr/bin/pamrubber
/usr/bin/pamscale
/usr/bin/pamseq
/usr/bin/pamshadedrelief
/usr/bin/pamsharpmap
/usr/bin/pamsharpness
/usr/bin/pamsistoaglyph
/usr/bin/pamslice
/usr/bin/pamsplit
/usr/bin/pamstack
/usr/bin/pamstereogram
/usr/bin/pamstretch
/usr/bin/pamstretch-gen
/usr/bin/pamsumm
/usr/bin/pamsummcol
/usr/bin/pamtable
/usr/bin/pamthreshold
/usr/bin/pamtilt
/usr/bin/pamtoavs
/usr/bin/pamtodjvurle
/usr/bin/pamtofits
/usr/bin/pamtogif
/usr/bin/pamtohdiff
/usr/bin/pamtohtmltbl
/usr/bin/pamtojpeg2k
/usr/bin/pamtompfont
/usr/bin/pamtooctaveimg
/usr/bin/pamtopam
/usr/bin/pamtopdbimg
/usr/bin/pamtopfm
/usr/bin/pamtopng
/usr/bin/pamtopnm
/usr/bin/pamtosrf
/usr/bin/pamtosvg
/usr/bin/pamtotga
/usr/bin/pamtotiff
/usr/bin/pamtouil
/usr/bin/pamtowinicon
/usr/bin/pamtoxvmini
/usr/bin/pamundice
/usr/bin/pamunlookup
/usr/bin/pamvalidate
/usr/bin/pamwipeout
/usr/bin/pamx
```

So, for sure I am missing some package or something that installs all these binaries... do you have any idea which package can it be?

This is for SLES:
```
susetest:~ # zypper se -i pam
S | Name       | Summary                                                       | Type
--+------------+---------------------------------------------------------------+--------
i | pam        | A Security Tool that Provides Authentication for Applications | package
i | pam-config | Modify common PAM configuration files                         | package
i | yast2-pam  | YaST2 - PAM Agent                                             | package 
```

And this is for JeOS:
```
susetest:~ # zypper se -i pam
S | Name       | Summary                                                       | Type
--+------------+---------------------------------------------------------------+--------
i | pam        | A Security Tool that Provides Authentication for Applications | package
i | pam-config | Modify common PAM configuration files                         | package
```

So, I have tried to install `yast2-pam` package in JeOS, but that's not the solution, those binaries are still missing... 
After some research, I found out that those binaries come from `netpbm` package that comes pre-installed in SLES but not in JeOS. 
```
susetest:~ # zypper se netpbm
S | Name        | Summary                                                      | Type
--+-------------+--------------------------------------------------------------+--------
i | libnetpbm11 | Libraries for the NetPBM (NetPortableBitmap) Graphic Formats | package
i | netpbm      | A Powerful Graphics Conversion Package                       | package
```


What I don't really understand is, why do we do use those binaries from a package that is about `A Powerful Graphics Conversion Package`?

**Comment by jlausuch** [comment]
TW: https://openqa.opensuse.org/tests/2259130
15-SP4: https://openqa.suse.de/tests/8369904

**Comment by lilyeyes** [comment]
For question “why do we do use those binaries ?", no special reason  I just found there are enough "pam*" files can be chosen to use.

**Review by lilyeyes — APPROVED** [review]
LGTM.

**Comment by jlausuch** [comment]
> For question “why do we do use those binaries ?", no special reason I just found there are enough "pam*" files can be chosen to use.

Fair enough, thanks! :)

File: tests/security/apparmor/aa_prepare.pm [hunk 1]
```diff
@@ -15,11 +15,16 @@ use strict;
...
 use utils 'zypper_call';
+use version_utils qw(is_jeos);
 use services::apparmor;
 
 sub run {
     select_console 'root-console';
     zypper_call 'in -t pattern apparmor';
+    if (is_jeos) {
+        record_info 'JeOS', 'Some packages needed by the tests are not pre-installed by default in JeOS.';
+        zypper_call 'in apparmor-utils screen nscd netpbm';
```
Comment (mloviska) [inline tests/security/apparmor/aa_prepare.pm:26]: `screen` ? Is it used in consequent tests ?

**Review by pdostal — APPROVED** [review]
Good for me except Martin's question :smile:

File: tests/security/apparmor/aa_prepare.pm (hunk 1 above)
Comment (jlausuch) [reply tests/security/apparmor/aa_prepare.pm:26]: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/security/apparmor/aa_genprof.pm#L40

**Pull request merged**